action.yaml

```diff
@@ -0,0 +1,117 @@
+      use `bash -l {0}`.
+      If you are using pip, you should use `bash`,
+      as `bash -l {0}` can fail on Mac runners.
+    required: true
+    default: bash
+outputs:
+  installed-version:
+    description: Installed MDAnalysis version
+    value: "${{ steps.print-version.outputs.installed-version }}"
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: Uninstall pre-existing versions
+      shell: ${{ inputs.shell }}
+      run: |
+        # uninstall tests even if not reinstalling,
+        # as versions can clash
+        python -m pip uninstall -y MDAnalysis MDAnalysisTests
+
+    - name: Set installer
+      id: setup-installer
+      shell: ${{ inputs.shell }}
+      run: |
+        case ${{ inputs.installer }} in
+            pip|conda|mamba) true ;;
+            *)
+              printf "${{ inputs.installer }} is not a valid installer; must be one of pip, conda, mamba"
+              exit 1
+              ;;
+        esac
+
+        INSTALLER="${{ inputs.installer }}"
+        if [[ ${{ inputs.version }} == "develop" ]] ; then
+          INSTALLER="pip"
+        fi
+
+        echo "::set-output name=installer::$(echo $INSTALLER)"
+
+
+    - name: Setup install lines
+      id: setup-install-line
+      shell: ${{ inputs.shell }}
+      run: |
+        if [[ "${{ inputs.version }}" == "latest" ]] ; then
+          _PACKAGE_SUFFIX=""
+        else
+          _PACKAGE_SUFFIX="==${{ inputs.version }}"
+        fi
+        
+        _INSTALL_LINE="MDAnalysis${_PACKAGE_SUFFIX}"
+        if [[ "${{ inputs.install-tests }}" == "true" ]] ; then
+          _INSTALL_LINE="${_INSTALL_LINE} MDAnalysisTests${_PACKAGE_SUFFIX}"
+        fi
+
+        # special case develop
+        if [[ "${{ inputs.version }}" == "develop" ]] ; then
+          _INSTALL_LINE="'git+https://github.com/MDAnalysis/mdanalysis@develop#egg=MDAnalysis&subdirectory=package'"
+          if [[ "${{ inputs.install-tests }}" == "true" ]] ; then
+            _INSTALL_LINE="${_INSTALL_LINE} 'git+https://github.com/MDAnalysis/mdanalysis@develop#egg=MDAnalysisTests&subdirectory=testsuite'"
+          fi
+        fi
+
+        echo "Install line: ${_INSTALL_LINE}"
+        echo "::set-output name=install-line::$(echo $_INSTALL_LINE)"
+    
+    - name: Install with conda
+      if: steps.setup-installer.outputs.installer != 'pip'
+      shell: ${{ inputs.shell }}
+      run: |
+        ${{ inputs.installer }} install -y -c conda-forge ${{ steps.setup-install-line.outputs.install-line }}
+        
+        if [[ ${{ inputs.version }} == "latest" ]] ; then
+          ${{ inputs.installer }} update -y -c conda-forge ${{ steps.setup-install-line.outputs.install-line }}
+        fi
+
+    - name: Install with pip
+      if: steps.setup-installer.outputs.installer == 'pip'
+      shell: ${{ inputs.shell }}
+      run: |
+        python -m pip install ${{ steps.setup-install-line.outputs.install-line }}
+
+        if [[ ${{ inputs.version }} == "latest" ]] ; then
```

Comment:
I'm not sure why this special case exists, I thought you could just straight up pip install from git develop?

Reply:
"develop" pulls from git develop -- "latest" pulls the latest release available on pypi/conda. Is that what you're asking?

Reply:
Ohhhh, calling upgrade again? I think because I have murky memories in the past about having to manually --upgrade to get the latest version, although I never nailed down exactly why.

Reply:
We pip install from main in a few places in OpenFE and I've never had this problem. Would it be worth seeing if it's not worth it?

Reply:
I'm confused what you're asking here -- are you asking why `pip --upgrade`, or are you asking to install from github? I was answering the former question. We could install from github for `latest` but I guess I don't quite see why -- especially since I assume we're renaming `master` at some point?

Reply:
Sorry my sole question here is about upgrade step overall and whether or not it's necessary to run it.